### DISCUSSION TITLE
How to replace default ICreatesObservable?

### DISCUSSION THREAD

**Discussion by glen-nicol** [reactiveui/ReactiveUI#3873] [Q&A] [answered]
How would one go about replacing the registration for the ICreatesObservable interface according to this [comment](https://github.com/reactiveui/ReactiveUI.Validation/issues/237#issuecomment-797817647)? I would like to experiment using the default WPF bindings to inherit the default WPF validation binding behavior. If not outright replacement, is there a way to add another instance that is including in the affinity voting?

I really like how RXUI bindings are refactor friendly and don't like having to revert to XAML bindings for any property with validation attached to it.

**Comment by ChrisPulman** [comment]
Have you tried using ReactiveUI.Validation?
https://www.reactiveui.net/docs/handbook/user-input-validation#reactiveuivalidation

ReactiveUI ReactiveProperty also has the ability to use validation,  I haven't documented the ReactiveProperty yet but will try to do this in the coming weeks.

**Reply by glen-nicol** [reply]
Hello again Chris! The problem with the validation package (and all provided RXUI code bindings) is it can't bind to attached properties and my control templates use the built in Validation.Errors attached property to display errors for the specific input control.

EDIT: I see that the Validation.Errors property isn't simple to work with even if RXUi could bind to attached properties. But this really is a big hole when integrating with other control libraries that assume Validation.Errors usage.

**Reply by ChrisPulman** [reply]
Do you have a basic repo of what you are trying to do i.e. a Wpf App with a basic example of an Attached Property that shows where you are trying to apply the validation; a zip of the code here would suffice.

**Reply by glen-nicol** [reply]
Unfortunately, Validation.Errors is set by the WPF binding code and its not a straight forward property that can just be set. But here is a repro of behavior I would like using WPF bindings. Would prefer to be able to use RXUI's functions to accomplish it though so I don't have to have two different styles of bindings.

[ReactiveValidationRepro.zip](https://github.com/user-attachments/files/16535712/ReactiveValidationRepro.zip)

**Reply by ChrisPulman** [reply, marked as answer]
I have created a PR that hopefully will be accepted, its Wpf only at present, but if accepted I will see what other platforms I can make this work with.
[Validation Binding](https://github.com/reactiveui/ReactiveUI/pull/3874)

**Reply by glen-nicol** [reply]
Wow thanks for the quick response Chris! Is there a reason you decided to require ReactiveProperty? Personally I like that the existing RXUI bindings don't require the viewmodel to use RXUI types to work. It allows it to work with legacy code and other libraries. Looking at your commit, I think the source of the binding could be changed to the viewmodel instance and then the path just needs to be extracted from the expression used to select the viewmodel property. I can make comments on the PR if you would prefer.

**Reply by ChrisPulman** [reply]
I will try to create a second non ReactiveProperty version, the key thing is to avoid reflection otherwise the Validation element is detached. 
The version created compliments the ReactiveProperty, so I will leave that one to stand. 
I will create a second PR for a version that doesn't target a ReactiveProperty.

**Reply by ChrisPulman** [reply]
I had to change Bind to BindWithValidation due to conflict with existing Bind methods

**Reply by ChrisPulman** [reply]
This additional binding will be included in the next Release, currently only supporting Wpf, no binding converters can be used with this Binding due to the reflection required.

**Reply by glen-nicol** [reply]
Chris you're awesome! I'll be on the lookout for those. Does that package have any prereleases I can use to try it out?

**Reply by ChrisPulman** [reply]
We're waiting for a new signing certificate before we can make any releases. Hopefully this will be completed in the next few days.
